client.go

```diff
@@ -603,6 +615,12 @@ func (c *Client) execute(req *Request) (*Response, error) {
 
 	req.Time = time.Now()
 	c.httpClient.Transport = c.transport
+
+	if req.RawRequest.URL != nil && req.RawRequest.URL.Scheme == "" {
+		req.RawRequest.URL.Scheme = c.scheme
+		req.RawRequest.URL.Host = req.URL
```

Comment:
Thanks, so you need only `Host` here correct?

If yes `req.URL` has complete request URL, you may need to parse `Host` from `req.URL`. Please correct me if I missing something.